base/workerpool.jl

```diff
@@ -114,3 +117,57 @@ using `remotecall_fetch`.
 """
 remote(f) = (args...; kwargs...)->remotecall_fetch(f, default_worker_pool(), args...; kwargs...)
 remote(p::WorkerPool, f) = (args...; kwargs...)->remotecall_fetch(f, p, args...; kwargs...)
+
+type CachedWorkerPool # Manages a cache for code/data on workers.
+    pool::WorkerPool
+    directory::Dict{Tuple, RemoteChannel}  # Mapping between a tuple (worker_id, object_id) and a remote_ref
+    CachedWorkerPool(pool) = new(pool, Dict{Int, Any}())
+end
+
+function release(p::CachedWorkerPool)  # For a quick release of remote function cache
```

Comment:
There is no guarantee on when a finalizer is called. It can especially hurt if the mapping closure is large (as in embedded data size) and there is a delay on it being cleared on the remote side. We can register it as a finalizer but will still need to call it before exiting the pmap call to ensure a quick release.